source/extensions/common/tap/extension_config_base.cc

```diff
@@ -65,9 +65,11 @@ void ExtensionConfigBase::installNewTap(const envoy::config::tap::v3::TapConfig&
                                         Sink* admin_streamer) {
   TapConfigSharedPtr new_config =
       config_factory_->createConfigFromProto(proto_config, admin_streamer);
-  tls_slot_.runOnAllThreads([new_config](OptRef<TlsFilterConfig> tls_filter_config) {
-    tls_filter_config->config_ = new_config;
-  });
+  tls_slot_.runOnAllThreads(
+      [new_config](OptRef<TlsFilterConfig> tls_filter_config) {
+        tls_filter_config->config_ = new_config;
+      },
+      []() -> void { ENVOY_LOG(debug, "New tap installed on all workers."); });
```

Comment:
Out of curiosity, would it work to move the ENVOY_LOG() into the body of the first lambda, after line 70?

Reply:
nope.  runonAllThreads is async, and allows for a callOnComplete closure, which happens  after all the posted closures are actually run 
If we moved it out we'd have the same problem where we'd log after we called runOnAllThreads but potentially before those async posted callbacks had actually run on individual worker threads

Reply:
Oh! I see. Run on *all* threads. Gotcha.